<?xml version="1.0"?>
<launch>

  <!-- Get time reference from clock topic from bag file -->
  <param name="use_sim_time" value="true" />  

  <!-- Setup tf frames to make rviz happy  -->
  <node pkg="tf2_ros" type="static_transform_publisher" name="velodyne_broadcaster" args="0 0 2.1 0 0 0.25 base_footprint velodyne" />
  <node pkg="tf2_ros" type="static_transform_publisher" name="camera_broadcaster" args="0 0 2.1 0 0 0 base_footprint camera" />

  <!-- Run lidar_filter node -->
  <node pkg="golfcart_push" type="lidar_filter" name="lidar_filter" output="screen" >
    <remap from="input_points" to="/velodyne_points" /> 
  </node>

  <!-- Run camera_vision node -->
  <node pkg="golfcart_push" type="camera_vision" name="camera_vision" output="screen" >
    <remap from="input_images" to="/front_camera/image_rect_color" /> <!-- /front_camera/image_raw -->
  </node>
  
  <!-- Get robot model -->
  <!-- Parse the URDF model and store the result in the 'robot_description' parameter -->
  <param name="robot_description" command="$(find xacro)/xacro '$(find golfcart_push)/urdf/r3d3_visual.urdf.xacro'" />

  <!-- This node loads the 'robot_description' parameter, subscribes to the 'joint_states' topic,
       and publishes TF transforms for the robot's joints -->
  <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher" />

  <node pkg="golfcart_push" type="r3d3_joint_pub" name="r3d3_joint_pub" />

  <!-- Rviz config -->
  <node type="rviz" name="rviz" pkg="rviz" args="-d $(find golfcart_push)/rviz/svc.rviz"/>

  <!-- Play bag, use sim time, run on loop (Can't easily print to terminal if ran here)
  <node pkg="rosbag" type="play" name="player" output="screen" args="(add dash here)-clock /home/bdawg/grudz_ws/src/golfcart_push/bags/moving-002.bag -l"/>  -->

  <!-- Reconfigure server -->
  <node pkg="rqt_reconfigure" type="rqt_reconfigure" name="reconfig" />
  
</launch>
